packages/typescript-estree/src/node-utils.ts

```diff
@@ -335,14 +335,21 @@ export function isJSXToken(node: ts.Node): boolean {
  */
 export function getDeclarationKind(
   node: ts.VariableDeclarationList,
-): 'const' | 'let' | 'var' {
-  if (node.flags & ts.NodeFlags.Let) {
-    return 'let';
-  }
-  if (node.flags & ts.NodeFlags.Const) {
-    return 'const';
+): 'const' | 'let' | 'var' | 'using' | 'await using' {
+  switch (node.flags) {
+    case ts.NodeFlags.Using:
+      return 'using';
+    case ts.NodeFlags.AwaitUsing:
+      return 'await using';
```

Comment:
...huh. `ts.NodeFlags` is not a proper exponents-of-two enum:

```ts
enum NodeFlags {
    None = 0,
    Let = 1,
    Const = 2,
    Using = 4,
    AwaitUsing = 6,
    NestedNamespace = 8,
    Synthesized = 16,
    Namespace = 32,
    // ...
}
```

https://github.com/microsoft/TypeScript/pull/54505/files#diff-e9fd483341eea176a38fbd370590e1dc65ce2d9bf70bfd317c5407f04dba9560R796 isn't the first instance of TS adding a combination value to the enum, but it does mess with our logic.

This direct equality checking doesn't work when `node.flags` has additional values on it. For example, putting it inside an `async function` causes an `await using`'s `kind` to be `"const"` and a `using`'s `kind` to be `"var"`.

So missing at least these cases:
* `using` inside a `function`
* `using` inside an `async function`
* `await using` inside an `async function`

Reply:
I appreciate your clarification about the misunderstanding regarding the determination of NodeFlags 🙏 

As you rightly mentioned,  AwaitUsing corresponds to `Const | Using`. To address this, I'll be making the following adjustments:
- Before distinguishing`Const` or `Using`, I will compare `node.flags & ts.NodeFlags.AwaitUsing`(result of `&` operation) with `ts.NodeFlags.AwaitUsing`.
  - This comparison will return true only when the `kind` of the node is `await using`.
- If the comparison mentioned above is not applicable, it indicates that the `kind` of the node is not `await using`. In this case, I will individually evaluate the `Const` and `Using` flags using `&`.